x-pack/plugin/ent-search/qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/xpack/application/FullClusterRestartIT.java

```diff
@@ -105,13 +118,36 @@ private void assertDslDataRetention(String analyticsCollectionName) throws IOExc
         for (Object dataStreamObj : dataStreams) {
             ObjectPath dataStream = new ObjectPath(dataStreamObj);
             if (dataStreamName.equals(dataStream.evaluate("name"))) {
-                assertNull(dataStream.evaluate("ilm_policy"));
                 assertEquals(true, dataStream.evaluate("lifecycle.enabled"));
                 assertEquals("180d", dataStream.evaluate("lifecycle.data_retention"));
+                assertEquals("Data stream lifecycle", dataStream.evaluate("next_generation_managed_by"));
+                assertEquals(false, dataStream.evaluate("prefer_ilm"));
                 evaluatedNewDataStream = true;
             }
         }
         assertTrue(evaluatedNewDataStream);
+    }
 
+    private void assertDataStreamTemplateExists(String templateName) throws IOException {
+        assertDataStreamTemplateExists(templateName, null);
+    }
+
+    private void assertDataStreamTemplateExists(String templateName, Integer minVersion) throws IOException {
+        try {
+            Request getIndexTemplateRequest = new Request("GET", "_index_template/" + templateName);
+            Response response = client().performRequest(getIndexTemplateRequest);
+            assertOK(response);
+            if (minVersion != null) {
+                String pathToVersion = "index_templates.0.index_template.version";
+                ObjectPath indexTemplatesResponse = ObjectPath.createFromResponse(response);
+                assertEquals(minVersion, indexTemplatesResponse.evaluate(pathToVersion));
```

Comment:
Is it possible that in the future the index template gets upgraded to a version number > 3? If so, check a greater-or-equals-than relation instead.